Documentation/api.md

```diff
@@ -578,14 +592,26 @@ TLSConfig specifies TLS configuration parameters.
 
 | Field | Description | Scheme | Required |
 | ----- | ----------- | ------ | -------- |
-| caFile | The CA cert to use for the targets. | string | false |
-| certFile | The client cert file for the targets. | string | false |
-| keyFile | The client key file for the targets. | string | false |
+| caFile | Path to the CA cert in the Prometheus container to use for the targets. | string | false |
+| caSecret | Secret containing the CA cert to use for the targets. | *[v1.SecretKeySelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#secretkeyselector-v1-core) | false |
+| certFile | Path to the client cert file in the Prometheus container for the targets. | string | false |
+| certSecret | Secret containing the client cert file for the targets. | *[v1.SecretKeySelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#secretkeyselector-v1-core) | false |
+| keyFile | Path to the client key file in the Prometheus container for the targets. | string | false |
+| keySecret | Secret containing the client key file for the targets. | *[v1.SecretKeySelector](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#secretkeyselector-v1-core) | false |
 | serverName | Used to verify the hostname for the targets. | string | false |
 | insecureSkipVerify | Disable target certificate validation. | bool | false |
 
 [Back to TOC](#table-of-contents)
 
+## TLSConfigValidationError
```

Comment:
Can we filter this out?

Reply:
changed binary responsible for generating it to skip empty entries.